src/codegen/wasm.zig

```diff
@@ -55,15 +55,48 @@ pub fn genCode(buf: *ArrayList(u8), decl: *Decl) !void {
     // Reserve space to write the size after generating the code
     try buf.resize(5);
 
-    // Write the size of the locals vec
-    // TODO: implement locals
-    try leb.writeULEB128(writer, @as(u32, 0));
+    // Calculate the locals of the body
+    // We must calculate the occurance of each value type and emit the count of it
+    const tv = decl.typed_value.most_recent.typed_value;
+    const mod_fn = tv.val.castTag(.function).?.data;
+
+    var locals = std.AutoArrayHashMap(u8, u32).init(buf.allocator);
```

Comment:
This will never have more than 4 keys so it could really be a `[4]u32` and avoid any allocation. Not sure if the semantics of the hashmap are worth it here for readability or not. If you do stick with the hashmap I don't think you rely on the ordering so the normal non-array hashmap would be preferable.

Reply:
I actually had this initially (9a34d6c) but in hindsight, the hashmap didn't really offer more readability than I had hoped for.